### DISCUSSION TITLE
A Few Questions: Html Output, Viewing Results and Backends

### DISCUSSION THREAD

**Discussion by ted-kord** [google-research/dex-lang#649] [Q&A] [answered]
I have a few questions.
1. How do I view html outputs. I ran the 'sierpinski' example but I just got a reproduction of the source code and "html output"  at the end. 
2. Most of the examples I run just reproduce the code. Is this deliberate? How do I see results or better still, what am I misunderstanding about the whole process.
3. How do I select between the cuda backend and cpu and is there a multithreaded backend?

**Comment by apaszke** [comment, marked as answer]
1. You can use `dex web` to open up the script in our notebook interface and `dex script --outfmt html` to convert a script into an HTML file that embeds the outputs. If you run `dex script` then any `:html` pragma will print `<html output>` since it can't be displayed in a terminal.
2. Yes, `dex script` reproduces the whole script and prints the result of evaluating every top-level expression right after it. You can use `dex script --outfmt result-only` to change this behavior.
3. All commands accept a `--backend` flag. `llvm-cuda` is the GPU backend while `llvm-mc` is the multicore CPU backend. But both can be flaky at times because we don't have full CI for them. Please open an issue if they don't work for you.

**Reply by ted-kord** [reply]
Thank you. 
(1) and (2) work fine but not the GPU or multicore backend. I will open an issue now.
